internal/resolve/catalog.go

```diff
@@ -95,31 +96,23 @@ func (r *CatalogResolver) Resolve(ctx context.Context, ext *ocv1alpha1.ClusterEx
 			return compare.ByVersion(a, b)
 		})
 
-		thisBundle := packageFBC.Bundles[0]
-		if resolvedBundle != nil {
-			// Cases where we stick with `resolvedBundle`:
-			//  1. If `thisBundle` is deprecated and `resolvedBundle` is not
-			//  2. If `thisBundle` and `resolvedBundle` have the same deprecation status AND `resolvedBundle` is a higher version
-			if isDeprecated(thisBundle, thisDeprecation) && !isDeprecated(*resolvedBundle, resolvedDeprecation) {
-				return nil
-			}
-			if compare.ByVersion(*resolvedBundle, thisBundle) < 0 {
-				return nil
-			}
-		}
-		resolvedBundle = &thisBundle
+		resolvedBundles = append(resolvedBundles, &packageFBC.Bundles[0])
+		matchedCatalogs = append(matchedCatalogs, cat.GetName())
 		resolvedDeprecation = thisDeprecation
 		return nil
 	}); err != nil {
 		return nil, nil, nil, fmt.Errorf("error walking catalogs: %w", err)
 	}
 
-	if resolvedBundle == nil {
+	if len(resolvedBundles) != 1 {
 		errPrefix := ""
 		if installedBundle != nil {
 			errPrefix = fmt.Sprintf("error upgrading from currently installed version %q: ", installedBundle.Version)
 		}
 		switch {
+		case len(resolvedBundles) > 1:
+			slices.Sort(matchedCatalogs) // sort for consistent error message
+			return nil, nil, nil, fmt.Errorf("%smatching packages found in multiple catalogs: %v; please specify a catalog", errPrefix, matchedCatalogs)
```

Comment:
This error message needs work I think on second glance. Open to suggestions!

Reply:
`"matching bundles found in multiple catalogs"` followed by a list of the catalogs or the list of catalogs+versions

Reply:
No need for "please specify a catalog"

Reply:
Updating this but I'm keeping `packages` vocabulary vs `bundles` since it matches more closely with the other errors and I think is more directly descriptive of the issue + what the user provided in `packageName`.